services/venice-router/src/main/java/com/linkedin/venice/router/throttle/ReadRequestThrottler.java

```diff
@@ -206,7 +213,7 @@ protected StoreReadThrottler getStoreReadThrottler(String storeName) {
   private StoreReadThrottler buildStoreReadThrottler(String storeName, int currentVersion, long storeQuotaPerRouter) {
     String topicName = Version.composeKafkaTopic(storeName, currentVersion);
     Optional<PartitionAssignment> partitionAssignment;
-    if (routingDataRepository.containsKafkaTopic(topicName)) {
+    if (perStorageNodeThrottlerEnabled && routingDataRepository.containsKafkaTopic(topicName)) {
```

Comment:
There might be a bug in today's implementation since I only see 
```
routingDataRepository.subscribeRoutingDataChange
```
when store got changed: `handleStoreChanged`, and no subscription in function: `buildStoreReadThrottler`.
Can you double check?

Reply:
huh! thats bad.  may be thats why I saw after router bounce so many stores getting quota/router count quota without any partition. calculation. in that case do we need to the 3x, probably more buffer as I saw quota rejection on that.

Reply:
I think this will only affect per-router-per-storage-node quota, not per-router quota.

Reply:
yes, thats the default today.